pkg/action/archive.go

```diff
@@ -300,8 +263,12 @@ func extractArchiveToTempDir(ctx context.Context, path string) (string, error) {
 	}
 
 	for file := range extractedFiles {
-		if err := extractNestedArchive(ctx, tmpDir, file, extractedFiles); err != nil {
-			return "", fmt.Errorf("extract nested archive: %w", err)
+		ext := getExt(file)
+		if _, ok := archiveMap[ext]; ok {
+			fmt.Printf("found a nested archive %s", file)
+			if err := extractNestedArchive(ctx, tmpDir, file, extractedFiles); err != nil {
```

Comment:
It made way more sense to only focus on one archive file at a time within `extractNestedArchive` instead of having it handle the iteration.